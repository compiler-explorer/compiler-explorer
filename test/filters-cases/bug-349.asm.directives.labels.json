{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "foo():"
    },
    {
      "labels": [],
      "source": null,
      "text": "; start of function"
    },
    {
      "labels": [],
      "source": null,
      "text": "; framesize_regs:     2"
    },
    {
      "labels": [],
      "source": null,
      "text": "; framesize_locals:   0"
    },
    {
      "labels": [],
      "source": null,
      "text": "; framesize_outgoing: 0"
    },
    {
      "labels": [],
      "source": null,
      "text": "; framesize:          2"
    },
    {
      "labels": [],
      "source": null,
      "text": "; elim ap -> fp       4"
    },
    {
      "labels": [],
      "source": null,
      "text": "; elim fp -> sp       0"
    },
    {
      "labels": [],
      "source": null,
      "text": "; saved regs: R4"
    },
    {
      "labels": [],
      "source": null,
      "text": "        ; start of prologue"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 1
      },
      "text": "        PUSHM.W #1, R4"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 1
      },
      "text": "        MOV.W   R1, R4"
    },
    {
      "labels": [],
      "source": null,
      "text": "        ; end of prologue"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        MOV.B   #3, R12"
    },
    {
      "labels": [],
      "source": null,
      "text": "        ; start of epilogue"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 3
      },
      "text": "        POPM.W  #1, r4"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 3
      },
      "text": "        RET"
    }
  ],
  "labelDefinitions": {}
}
