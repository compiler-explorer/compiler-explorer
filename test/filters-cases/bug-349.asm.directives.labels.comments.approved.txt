{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "foo():"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 1
      },
      "text": "        PUSHM.W #1, R4"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 1
      },
      "text": "        MOV.W   R1, R4"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        MOV.B   #3, R12"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 3
      },
      "text": "        POPM.W  #1, r4"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 3
      },
      "text": "        RET"
    }
  ],
  "labelDefinitions": {}
}
